Solutions/Azure Activity/Analytic Rules/Granting_Permissions_To_Account_detection.yaml

```diff
@@ -33,37 +33,45 @@ query: |
   | where TimeGenerated between (ago(starttime) .. ago(endtime))
   | summarize count() by CallerIpAddress, Caller
   | where count_ >= alertOperationThreshold
+  // Returns all the records from the right side that don't have matches from the left.
   | join kind = rightanti ( 
   createRoleAssignmentActivity
   | where TimeGenerated > ago(endtime)
-  | extend PrincipalId = tostring(parse_json(tostring(parse_json(tostring(Properties_d.requestbody)).Properties)).PrincipalId)
-  | extend PrincipalType = tostring(parse_json(tostring(parse_json(tostring(Properties_d.requestbody)).Properties)).PrincipalType)
-  | extend Scope = tostring(parse_json(tostring(parse_json(tostring(Properties_d.requestbody)).Properties)).Scope)
-  | extend RoleAddedDetails = tostring(parse_json(tostring(parse_json(tostring(parse_json(Properties).requestbody)).Properties)).RoleDefinitionId) 
-  | summarize StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated), ActivityTimeStamp = make_set(TimeGenerated), ActivityStatusValue = make_set(ActivityStatusValue), 
-  OperationIds = make_set(OperationId), CorrelationId = make_set(CorrelationId), ActivityCountByCallerIPAddress = count()  
+  | extend PrincipalId = tostring(iff(Properties_d has_cs 'PrincipalId',parse_json(tostring(parse_json(Properties_d).requestbody)).Properties.PrincipalId,parse_json(tostring(parse_json(Properties_d).requestbody)).properties.principalId))
```

Comment:
It might be good to use Case here in order to gave a fall back if we don't have values to parse out ultimately.

Also you are doing parse_json(tostring(parse_json  - is this really necessary? Would not just parse_json(Propoerties_d).requestbody work?

Reply:
1. Updated to use 'case' logic
2. Yes, had to use parse_json twice as I was not able to get nested value after parse_json(Properties_d).requestbody. Our documentation shows usage of parse_json twice in the last example: https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/parsejsonfunction#example